### ISSUE TITLE
Support for Microsoft.LabServices/labplans subnet service delegation

### ISSUE THREAD

**Issue by Voxys** [hashicorp/terraform-provider-azurerm#18819]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The Microsoft.LabServices/labplans service delegation is not present as possible values for `name` argument in the documentation : https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/subnet#service_delegation

### New or Affected Resource(s)/Data Source(s)

azurerm_subnet

### Potential Terraform Configuration

```hcl
resource "azurerm_subnet" "terraform-labs-subnet" {
  name                 = "subnet1"
  resource_group_name  = azurerm_resource_group.terraform-labs-rg.name
  virtual_network_name = azurerm_virtual_network.terraform-labs-vn.name
  address_prefixes     = ["10.3.1.0/24"]

  delegation {
    name = "delegation"

    service_delegation {
      name = "Microsoft.LabServices/labplans"
    }
  }
}
```

**Comment by liuwuliuyun** [comment]
I have raised PR for this issue.

**Comment by stephybun** [comment]
Closed by #18822
